main/EDInitMod.F90

```diff
@@ -298,14 +313,67 @@ subroutine set_site_properties( nsites, sites,bc_in )
+
+          if(hlm_use_fixed_biogeog.eq.itrue)then  
+           ! MAPPING OF FATES PFTs on to HLM_PFTs
+            ! add up the area associated with each FATES PFT
+            ! where pft_areafrac is the area of land in each HLM PFT and (from surface dataset)
+            ! hlm_pft_map is the area of that land in each FATES PFT (from param file)
+
+            sites(s)%area_pft(1:numpft) = 0._r8
+            do hlm_pft = 1,size( EDPftvarcon_inst%hlm_pft_map,2)
+               do fates_pft = 1,numpft ! loop round all fates pfts for all hlm pfts        
+                 sites(s)%area_pft(fates_pft) = sites(s)%area_pft(fates_pft) + &
+                     EDPftvarcon_inst%hlm_pft_map(fates_pft,hlm_pft) * bc_in(s)%pft_areafrac(hlm_pft) 
+               end do
+            end do !hlm_pft
+
+            sumarea = sum(sites(s)%area_pft(1:numpft))
+            do ft =  1,numpft
+              if(sites(s)%area_pft(ft).lt.0.01_r8.and.sites(s)%area_pft(ft).gt.0.0_r8)then
+                write(fates_log(),*)  'removing small pft patches',s,ft,sites(s)%area_pft(ft)
+                 sites(s)%area_pft(ft)=0.0_r8 
+                 ! remove tiny patches to prevent numerical errors in terminate patches
+                endif
+           end do
+! change units to m2 from fractions
             do ft =  1,numpft
-              sites(s)%area_pft(ft) = bc_in(s)%pft_areafrac(ft)
+             sites(s)%area_pft(ft)= sites(s)%area_pft(ft) * AREA ! rescale units to m2. 
+             if(sites(s)%area_pft(ft).lt.0._r8)then
+               write(fates_log(),*) 'negative area',s,ft,sites(s)%area_pft(ft)
+               call endrun(msg=errMsg(sourcefile, __LINE__))
+             end if
             end do
-          end if
 
-          do ft = 1,numpft
+           ! re-normalize PFT area to ensure it sums to one.
+           ! note that in areas of 'bare ground' (PFT 0 in CLM/ELM)
+           ! the bare ground will no longer be proscribed and should emerge from FATES
+           ! this may or may not be the right way to deal with this?
+
+            if(hlm_use_sp.eq.ifalse)then ! when not in SP mode, subsume bare ground evenly into the existing patches. 
+            !n.b. that it might be better if nocomp mode used the same bare groud logic as SP mode. 
+              sumarea = sum(sites(s)%area_pft(1:numpft))
+              do ft =  1,numpft
+                if(sumarea.gt.0._r8)then
+                  sites(s)%area_pft(ft) = area * sites(s)%area_pft(ft)/sumarea
+                else
+                  sites(s)%area_pft(ft) = area/numpft 
+                  ! in nocomp mode where there is only bare ground, we assign equal area to 
+                  ! all pfts and let the model figure out whether land should be bare or not. 
+                end if
+               end do !ft
+              else ! for sp mode, assert a bare ground patch
+                sumarea = sum(sites(s)%area_pft(1:numpft)) 
+
+                if(sumarea.lt.area)then !make some bare ground
```

Comment:
It's because in all the other FATES modes, bareground is the area in which trees do not grow of their own accord. In SP mode, because we are asserting that there is a full canopy in each of the PFT patches, we also need to assert a bare ground area in order to not have all of the ground filled by leaves.

Further to that, one could imagine having bare ground being calculated as the remaining area when all fhe canopies are accounted for, but this means we don't pass balance check on canopy are inside FATES, and so in SP mode, we define the bare groud/no cohorts patch as having a PFT identifier as zero.  

I added comments to this effect to help people understand this a bit more.